### PR TITLE
[mme][refactor] Remove AssertFatal from mme_api_get_emm_config

### PR DESCRIPTION
**Pull request by andreilee**
Signed-off-by: Andrei Lee <andreilee@fb.com>

## Summary

This PR is among a series of changes to rework how the `AssertFatal` macro is used in MME. As part of this, API methods should not use `AssertFatal`, instead allowing callers higher up to decide whether or not to stop MME.

In this PR, function `mme_api_get_emm_config(..)` is modified to remove all calls to the `AssertFatal` macro, instead returning `RETURNerror`. The only caller of this function is `emm_main_initialize(..)`, which has been modified to call `Fatal`.

## Test Plan

Only built MME service:

```
vagrant@magma-dev:~/magma/lte/gateway$ make build_oai
.
.
.
[1971/1971] Linking CXX executable oai_mme/mme
vagrant@magma-dev:~/magma/lte/gateway$ 
```

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Review by ulaskozat — CHANGES_REQUESTED** [review]
We lose granularity on what config was the culprit for initialization failure.

**Comment by andreilee** [comment]
@ulaskozat MME should still throw, no? And it looks like there's only one config that gets processed by that function as well.

File: lte/gateway/c/core/oai/tasks/nas/emm/emm_main.c [hunk 1]
```diff
@@ -62,8 +63,7 @@ void emm_main_initialize(const mme_config_t* mme_config_p) {
    */
   memset(&_emm_data.conf, 0, sizeof(_emm_data.conf));
   if (mme_api_get_emm_config(&_emm_data.conf, mme_config_p) != RETURNok) {
-    OAILOG_ERROR(
-        LOG_NAS_EMM, "EMM-MAIN  - Failed to get MME configuration data");
+    Fatal("EMM-MAIN  - Failed to get MME configuration data");
```
Comment (ulaskozat) [inline lte/gateway/c/core/oai/tasks/nas/emm/emm_main.c]: Maybe add more context to message, e.g., "EMM-MAIN - Failed to get all required MME config data, check the error logs for missing configs"

**Review by ulaskozat — APPROVED** [review]
Okay I guess we are printing the Error logs so one can figure out the offending configs. I added a nit comment so that message provides hint to look for the missing config.

**Pull request merged**